Compliance: Paper wallet printing

littlenaut opened the pull request.

Commit:
Add HTTP method to create a new wallet on request

Commit:
Allow domain selection when generating certificates

Commit:
Remove temporary logging code

ghost commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Private key printing](https://trello.com/c/A9y5mQE1/808-private-key-printing)

joshmh commented on lib/routes.js:
```diff
@@ -155,6 +155,15 @@ function postTx (req, res, next) {
 
+function createWallet (req, res, next) {
+  const pi = plugins(req.settings, req.deviceId)
+  pi.newAddress(req.tx)
```
What about the private key?

Commit:
Add 'cash_in_txs' migration

This migration adds a new column to the 'cash_in_txs' table to
accommodate the new paper wallet feature for compliance.

Commit:
Remove wallet creation HTTP method

This functionality is now completely implemented at client-side (machine).

joshmh commented on lib/routes.js:
```diff
@@ -367,6 +367,7 @@ app.post('/phone_code', getCustomerWithPhoneCode)
 app.patch('/customer/:id', updateCustomer)
 
 app.post('/tx', postTx)
+app.post('/tx/createWallet', createWallet)
```
Where is the ``createWallet`` function?

littlenaut replied:
That one escaped me. Removed.

joshmh commented on migrations/1552064796155-tx-paper-wallet.js:
```diff
@@ -0,0 +1,15 @@
+
+exports.up = function (next) {
+  var sql = [
+    'ALTER TABLE cash_in_txs ADD COLUMN is_paper_wallet boolean null DEFAULT false',
```
Is this being used?

littlenaut replied:
Yes, the [migration script](https://github.com/lamassu/lamassu-server/blob/dev/bin/lamassu-migrate) goes through all the scrips in the migration directory.

joshmh replied:
I mean is the ``is_paper_wallet`` field ever being set?

littlenaut replied:
Yes, it is. [See here](https://github.com/lamassu/lamassu-machine/blob/449a6cae14d97aa875281cd72372bd040a554699/lib/brain.js#L1775).

Commit:
Remove unused 'createWallet' HTTP method
